app/services/FinancialDetailsService.scala

```diff
@@ -60,64 +61,58 @@ class FinancialDetailsService @Inject()(val financialDetailsConnector: Financial
     }
   }
 
-  @deprecated("Use getAllFinancialDetailsV2 instead", "MISUV-8845")
   def getAllFinancialDetails(implicit user: MtdItUser[_],
-                             hc: HeaderCarrier, ec: ExecutionContext): Future[List[(Int, FinancialDetailsResponseModel)]] = {
-    Logger("application").debug(
-      s"Requesting Financial Details for all periods for mtditid: ${user.mtditid}")
-
-    Future.sequence(user.incomeSources.orderedTaxYearsByYearOfMigration.map {
-      taxYear =>
-        Logger("application").debug(s"Getting financial details for TaxYear: ${taxYear}")
-        financialDetailsConnector.getFinancialDetails(taxYear, user.nino).map {
-          case financialDetails: FinancialDetailsModel => Some((taxYear, financialDetails))
-          case error: FinancialDetailsErrorModel if error.code != NOT_FOUND => Some((taxYear, error))
-          case _ => None
-        }
-    }).map(_.flatten)
-  }
-
-  def getAllFinancialDetailsV2(implicit user: MtdItUser[_],
-                               hc: HeaderCarrier, ec: ExecutionContext): Future[Option[FinancialDetailsResponseModel]] = {
+                             hc: HeaderCarrier, ec: ExecutionContext): Future[Option[FinancialDetailsResponseModel]] = {
     Logger("application").debug(
       s"Requesting Financial Details for all periods for mtditid: ${user.mtditid}")
 
     val yearsOfMigration = user.incomeSources.orderedTaxYearsByYearOfMigration
     if (yearsOfMigration.isEmpty)
       Future.successful(None)
     else {
-      val (from, to) = (yearsOfMigration.min, yearsOfMigration.max)
-      Logger("application").debug(s"Getting financial details for TaxYears: ${from} - ${to}")
-
-      for {
-        response <- financialDetailsConnector.getFinancialDetails(TaxYear.forYearEnd(from), TaxYear.forYearEnd(to), user.nino)
-      } yield response match {
-        case financialDetails: FinancialDetailsModel => Some(financialDetails)
-        case error: FinancialDetailsErrorModel if error.code != NOT_FOUND => Some(error)
-        case _ => None
-      }
-    }
-  }
-
-  @deprecated("Use getUnpaidFinancialDetailsV2 instead", "MISUV-8845")
-  def getAllUnpaidFinancialDetails()(implicit user: MtdItUser[_], hc: HeaderCarrier, ec: ExecutionContext): Future[List[FinancialDetailsResponseModel]] = {
-    getAllFinancialDetails.map { chargesWithYears =>
-      chargesWithYears.flatMap {
-        case (_, errorModel: FinancialDetailsErrorModel) => Some(errorModel)
-        case (_, financialDetails: FinancialDetailsModel) =>
-          val unpaidDocDetails: List[DocumentDetail] = financialDetails.unpaidDocumentDetails()
-          if (unpaidDocDetails.nonEmpty) Some(financialDetails.copy(documentDetails = unpaidDocDetails)) else None
-      }
+      val maxYears = 5
```

Comment:
we might need to move this value to the level of FrontendAppConfig, still not sure if we need to make this configurable